src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php

```diff
@@ -138,6 +138,23 @@ public function registerProvider(ExpressionFunctionProviderInterface $provider)
     protected function registerFunctions()
     {
         $this->addFunction(ExpressionFunction::fromPhp('constant'));
+
+        $this->addFunction(new ExpressionFunction('enum',
+            static fn ($str): string => sprintf('\constant(%1$s)', $str),
+            static function ($arguments, $str): \UnitEnum {
+                if (!\defined($str)) {
+                    throw new \LogicException(sprintf('The enum case "%s" is not defined.', $str));
+                }
+
+                $value = \constant($str);
+
+                if (!$value instanceof \UnitEnum) {
+                    throw new \LogicException(sprintf('The string "%s" is not the name of a valid enum case.', $str));
```

Comment:
this should be an Error, or a TypeError maybe
BUT, doing this check is creating a discrepancy with the compiled version, which doesn't enforce the type of the result
we must ensure that both compiled and non-compiled variants behave the same

Reply:
I think the main point of using `enum()` function is to have this type check. I updated the code, so both evaluation and compilation behaves the same :+1: